core/src/main/scala/org/broadinstitute/workbench/hamm/model/model.scala

```diff
@@ -414,8 +424,16 @@ final case class Call(runtimeAttributes: RuntimeAttributes,
                       backend: BackEnd,
                       attempt: Attempt)
 
-final case class MetadataResponse(calls: List[Call], startTime: Instant, endTime: Instant)
+final case class MetadataResponse(calls: List[Call], startTime: Instant, endTime: Instant, workflowCollectionId: WorkflowCollectionId, labels: Map[String, String])
 
+object MetadataResponse {
+  def apply(calls: List[Call], startTime: Instant, endTime: Instant, labels: Map[String, String]): MetadataResponse = {
+    labels.get("caas-collection-name") match {
+      case Some(c) => MetadataResponse(calls, startTime, endTime, WorkflowCollectionId(UUID.fromString(c)), labels)
+      case None => throw new Exception(s"Workflow did not have a collection associated with it.")
```

Comment:
this seems kinda weird pattern since the case class already gives an `apply` method. What's the gain with overloading `apply`?

 If you need to do this labels check, can you just do it in https://github.com/DataBiosphere/hamm/blob/master/core/src/main/scala/org/broadinstitute/workbench/hamm/model/CromwellMetadataJsonCodec.scala#L87 ? That way, you don't need to throw an exception either.